lib/streamlit/media_file_manager.py

```diff
@@ -58,8 +63,19 @@ def _calculate_file_id(data, mimetype):
     return filehash.hexdigest()
 
 
+def _get_extension_for_mimetype(mimetype: str) -> str:
+    try:
```

Comment:
I think we should add a comment based on your observations that different Python version indicate a different MimeType. It would explain the need for a dictionary and checking it.